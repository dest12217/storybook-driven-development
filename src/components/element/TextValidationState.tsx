import * as React from 'react';

export default function TextValidationState({ children }: React.PropsWithChildren) {
  return <>
    <style href="element-text-validation-state" precedence="medium">{`
      .element-text-validation-state {
        color: #bf1650;
      }

      .element-text-validation-state::before {
        display: inline;
        content: "âš  ";
      }
    `}</style>
    <p className="element-text-validation-state" aria-live="assertive">{children}</p>
  </>;
}
