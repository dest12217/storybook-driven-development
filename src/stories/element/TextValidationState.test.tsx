import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import TextValidationState from '@/components/element/TextValidationState';

describe('TextValidationState', () => {
  test('Snapshot', () => {
    const { asFragment } = render(<TextValidationState />);

    expect(asFragment()).toMatchSnapshot();
  });
});
