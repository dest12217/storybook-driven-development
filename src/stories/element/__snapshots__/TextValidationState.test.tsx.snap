// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextValidationState > Snapshot 1`] = `
<DocumentFragment>
  <p
    aria-live="assertive"
    class="element-text-validation-state"
  />
</DocumentFragment>
`;
